docs/src/tutorial.rst

```diff
@@ -0,0 +1,487 @@
+    const S3 = new AWS.S3();
+
+    const bucketName = process.env.BUCKET;
+
+    // For now we only handle GET /:
+    exports.main = async function(event, context) {
+      try {
+        var method = event.httpMethod;
+
+        if (method == "GET") {
+          if (event.path.length == 1) {
+            const data = await S3.listObjectsV2({ Bucket: bucketName }).promise();
+            var body = {
+              widgets: data.Contents.map(function(e) { return e.Key })
+            };
+            return {
+              statusCode: 200,
+              headers: {},
+              body: JSON.stringify(body)
+            };
+          }
+          else {
+            // GET /name
+            var name = event.path.substring(1, event.path.length);
+  
+            const data = await S3.getObject({ Bucket: bucketName, Key: name}).promise();
+            var body = data.Body.toString('utf-8');
+
+            return {
+              statusCode: 200,
+              headers: {},
+              body: JSON.stringify(body)
+            };
+          }
+        }
+        else {
+          var body = "We only accept GET, POST, and DELETE, not " + method;
+        
+          return {
+            statusCode: 200,
+            headers: {},
+            body: JSON.stringify(body)
+          };
+        }
+      } catch(error) {
+        var body = error.stack || JSON.stringify(error, null, 2);
+        var response = makeError(body);
+        return response;
+      }
+    }
+
+Save it and make sure it builds and creates an empty stack.
+Note that since we haven't wired the function to our app,
+the Lambda file does not appear in the output.
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
+
+.. _create_widgets_service:
+
+Step 3: Create Widgets Service
+==============================
+
+Add the |APIGATEWAY|, |LAM|, and |S3| packages to our app.
+
+.. code-block:: sh
+
+    npm install @aws-cdk/aws-apigateway @aws-cdk/aws-lambda @aws-cdk/aws-s3
+
+Create the directory *lib* at the same level as the *bin* and *resources*
+directories.
+
+.. code-block:: sh
+
+    mkdir lib
+
+Create the following Typescript file, *widget_service.ts*,
+in the *lib* directory.
+
+.. code-block:: ts
+
+    import cdk = require('@aws-cdk/cdk');
+    import apigateway = require('@aws-cdk/aws-apigateway');
+    import lambda = require('@aws-cdk/aws-lambda');
+    import s3 = require('@aws-cdk/aws-s3');
+
+    export class WidgetService extends cdk.Construct {
+      constructor(parent: cdk.Construct, name: string) {
+        super(parent, name);
+
+        // Use S3 bucket to store our widgets
+        const bucket = new s3.Bucket(this, 'WidgetStore');
+
+        // Create a handler that calls the function main
+        // in the source file widgets(.js) in the resources directory
+        // to handle requests through API Gateway
+        const handler = new lambda.Function(this, 'WidgetHandler', {
+          runtime: lambda.Runtime.NodeJS610,
```

Comment:
You switched your code to using `async`/`await` (better!) but you also have to update the runtime to use a NodeJS version of 8 or higher, otherwise it's not going to work.

Is the code tested?

Reply:
I plan to replace the code in the file with literalinclude directives so I can test the code.

Currently the code is in separate files as suggested by the doc.